src/metabase/driver/common/parameters/values.clj

```diff
@@ -201,9 +201,18 @@
                                       (parse-value-for-type param-type v))})
 
     (and (= param-type :dimension)
-         (get-in value [:field :base_type])
-         (string? (get-in value [:value :value])))
-    (update-in value [:value :value] (partial parse-value-for-field-base-type (get-in value [:field :base_type])))
+         (get-in value [:field :base_type]))
+    (cond
```

Comment:
Would you mind factoring out this form into a separate function? Maybe call it something like `parse-value-for-dimension`, and taking args for only `value` and `base-type` (to avoid the repeated look-ups).

Reply:
I went with the `parse-value-for-dimension` approach that you suggested, but ended up in a situation where I'd have to return `nil` and handle that in `parse-value-for-type` anyway. I'm still avoiding repeated lookups, but acting on the field filter as a whole rather than just the `base-type` and `value`.